File: src/main/java/com/example/chatService/entity/DMMessage.java
package com.example.chatService.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.OffsetDateTime;

@Entity
@Table(name = "dm_messages")
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class DMMessage {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "room_id", referencedColumnName = "roomId")
    private DMRoom room;

    @Column(length = 36, nullable = false)
    private String senderId;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    @Builder.Default
    private OffsetDateTime sentAt = OffsetDateTime.now();

    @Column(name = "is_read")
    @Builder.Default
    private boolean isRead = false;
}

File: src/main/java/com/example/chatService/entity/DMMessage.kt
package com.example.chatService.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import java.time.OffsetDateTime

@Entity
@Table(name = "dm_messages")
class DMMessage(

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long? = null,

        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "room_id", referencedColumnName = "roomId")
        var room: DMRoom? = null,

        @Column(length = 36, nullable = false)
        var senderId: String? = null,

        @Column(nullable = false, columnDefinition = "TEXT")
        var content: String? = null,

        var sentAt: OffsetDateTime = OffsetDateTime.now(),

        @Column(name = "is_read")
        var isRead: Boolean = false
)